tests/acl/test_stress_acl.py

```diff
@@ -24,17 +27,105 @@
 
 LOG_EXPECT_ACL_TABLE_CREATE_RE = ".*Created ACL table.*"
 LOG_EXPECT_ACL_RULE_FAILED_RE = ".*Failed to create ACL rule.*"
 
 ACL_RULE_NUMS = 10
 
+DEFAULT_MAX_ACL_ENTRIES = 200
+ARISTA_7050_MAX_ACL_ENTRIES = 512
+
+# key: hwsku value as per 'show version | grep -i hwsku'
+# value: max number of ACL entries supported by the platform
+rules_per_hwsku = {
+    'Arista-7050CX3-32S-C32': ARISTA_7050_MAX_ACL_ENTRIES,
+    'Arista-7050-QX-32S': ARISTA_7050_MAX_ACL_ENTRIES,
+    'Arista-7050CX3-32S-C32': ARISTA_7050_MAX_ACL_ENTRIES,
+    'Arista-7050CX3-32S-D48C8': ARISTA_7050_MAX_ACL_ENTRIES,
+    'Arista-7050QX32S-Q32': ARISTA_7050_MAX_ACL_ENTRIES,
+    'Arista-7060CX-32S-C32': ARISTA_7050_MAX_ACL_ENTRIES,
+    'Arista-7060CX-32S-D48C8': ARISTA_7050_MAX_ACL_ENTRIES,
+    'Arista-7060CX-32S-Q32': ARISTA_7050_MAX_ACL_ENTRIES,
+    'Arista-7060DX5-32': ARISTA_7050_MAX_ACL_ENTRIES,
+    'Arista-7060X6-64PE-256x200G': ARISTA_7050_MAX_ACL_ENTRIES,
+    'Arista-7060X6-64PE-C224O8': ARISTA_7050_MAX_ACL_ENTRIES,
+    'Arista-7060X6-64PE-C256S2': ARISTA_7050_MAX_ACL_ENTRIES,
+    'Arista-7170-64C': ARISTA_7050_MAX_ACL_ENTRIES,
+    'Arista-720DT-G48S4': ARISTA_7050_MAX_ACL_ENTRIES,
+    'Arista-7260CX3-C64': ARISTA_7050_MAX_ACL_ENTRIES,
+    'Arista-7260CX3-D108C10': ARISTA_7050_MAX_ACL_ENTRIES,
+    'Arista-7260CX3-D108C8': ARISTA_7050_MAX_ACL_ENTRIES,
+    'Arista-7280CR3-C40': ARISTA_7050_MAX_ACL_ENTRIES,
+    'Arista-7800R3-48CQ2-C48': ARISTA_7050_MAX_ACL_ENTRIES,
+    'Arista-7800R3-48CQM2-C48': ARISTA_7050_MAX_ACL_ENTRIES,
+    'Arista-7800R3A-36DM2-C36': ARISTA_7050_MAX_ACL_ENTRIES,
+    'Arista-7800R3A-36DM2-D36': ARISTA_7050_MAX_ACL_ENTRIES,
+    'Arista-7808R3A-FM': ARISTA_7050_MAX_ACL_ENTRIES
+}
+
+
+@pytest.fixture(scope="module")
+def setup_table_and_rules(rand_selected_dut, prepare_test_port):
+
+    logger.debug('Setting up rules')
+    _, _, dut_port = prepare_test_port
+    logger.debug(f'dut_port: {dut_port}')
+    table_name = 'STRESS_ACL_MANY'
+
+    # Get the max number of ACL entries supported by the platform
+    hwsku = rand_selected_dut.facts['hwsku']
+    max_acl_entries = rules_per_hwsku.get(hwsku, DEFAULT_MAX_ACL_ENTRIES)
+    logger.debug(f'Max ACL entries supported by the platform: {max_acl_entries}')
+    rules = generate_acl_rules(table_name, max_acl_entries)
+    f_name = f'generated_acl_rules_{max_acl_entries}.json'
+    file_path = f'/tmp/{f_name}'
+    with open(file_path, 'w') as f:
+        json.dump(rules, f)
+    logger.debug(f'Generated ACL rules written to {file_path}')
+    # Add table
+    cmd_add_table = f'config acl add table {table_name} L3 -s ingress -p {dut_port}'
+    rand_selected_dut.shell(cmd_add_table)
+    logger.debug('Table created')
+    # Copy rules file and add rules
+    rand_selected_dut.copy(src=file_path, dest=f'/tmp/{f_name}', mode="0755")
+    cmd_add_rules = f'sonic-cfggen -j /tmp/{f_name} -w'
+    rand_selected_dut.shell(cmd_add_rules)
```

Comment:
Please add a check to ensure the status of all the rules is active. If any rule is active, then no need to run the traffic test.

Reply:
I have changed the code to check for active rules. Test runs only if number of active rules match number of input rules.